### PR TITLE
android: enable std::bind

### PR DESCRIPTION
**Pull request by andyLiuChina**
define BIND as std::bind which declare in c++11,
so we should add compiling option -std=c++11.

### CODE REVIEW

**Comment by xuguangxin** [comment]
+1

File: interface/VideoCommonDefs.h [hunk 1]
```diff
@@ -33,12 +33,14 @@
...
 namespace YamiMediaCodec{
 #ifdef ANDROID
+    #define BIND std::bind
     #define SharedPtr std::shared_ptr
     #define WeakPtr std::weak_ptr
     #define DynamicPointerCast std::dynamic_pointer_cast
     #define StaticPointerCast std::static_pointer_cast
     #define EnableSharedFromThis std::enable_shared_from_this
 #else
+    #define BIND std::tr1::bind
```
Comment (uartie) [inline interface/VideoCommonDefs.h]: Why all uppercase?  It's inconsistent with the rest of the naming.  How about "Bind" instead?

File: interface/VideoCommonDefs.h [hunk 2]
```diff
@@ -33,12 +33,14 @@
 #endif
 namespace YamiMediaCodec{
 #ifdef ANDROID
+    #define BIND std::bind
```
Comment (uartie) [inline interface/VideoCommonDefs.h]: std::bind is defined in <functional> and should be included.

File: interface/VideoCommonDefs.h (hunk 1 above)
Comment (uartie) [inline interface/VideoCommonDefs.h]: std::tr1::bind is defined in <tr1/functional> and should be included

**Comment by xuguangxin** [comment]
+1

**Comment by xuguangxin** [comment]
merged as 643017287b8258fe4566f6828977efc5a9cd2950

**Pull request closed without merging**